c++/src/kj/compat/http.c++

```diff
@@ -5709,17 +5709,36 @@ public:
       }).attach(kj::mv(address));
     }).split();
 
+    auto connection = kj::refcountedWrapper(kj::newPromisedStream(kj::mv(kj::get<1>(split))));
```

Comment:
Nit: the overhead of the refcounted wrapper isn't needed if the conditionals below aren't taken.